### PR TITLE
Add support for Plausible analytics and Matomo analytics

### PR DESCRIPTION
**Pull request by selten**
Adds #10954, allowing GDPR compliance as well as self-hosting the analytics systems.

### CODE REVIEW

File: lib/galaxy/config/sample/galaxy.yml.sample [hunk 1]
```diff
@@ -817,6 +817,14 @@ galaxy:
...
   #ga_code: null
 
+  # Please enter the URL for the Plausible server (including https) so this can be used for tracking
+  # with Plausible (https://plausible.io/).
+  # plausible_server: null
+
+  # Please enter the URL for the Galaxy server so this can be used for tracking
+  # with Plausible (https://plausible.io/).
+  #plausible_domain: null
+
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/galaxy.yml.sample:828]: This is missing the matomo options, can you do a `make config-rebuild` and add the changes (also the ones in `doc/source/admin/` ?

Comment (selten) [reply lib/galaxy/config/sample/galaxy.yml.sample:828]: Thanks for pointing that out! I didn't know those commands

File: templates/galaxy_client_app.mako [hunk 2]
```diff
@@ -71,6 +71,43 @@ ${ h.dumps( dictionary, indent=( 2 if trans.debug else 0 ) ) }
     </script>
 </%def>
 
+<%def name="config_plausible_analytics(plausible_server, plausible_domain)">
+    <script>
+        console.log("config_plausible_analytics plausible_server:", '${plausible_server}');
+        console.log("config_plausible_analytics plausible_domain:", '${plausible_domain}');
+    </script>
```
Comment (hexylena) [inline templates/galaxy_client_app.mako]: is this necessary? if not might remove.

Comment (selten) [reply templates/galaxy_client_app.mako]: It's not necessary, just thought I'd keep the same logging as Google Analytics has.

**Review by hexylena — APPROVED** [review]
Works well for me!

image

**Comment by hexylena** [comment]
```
FAILED test/unit/tools/test_toolbox.py::ToolBoxTestCase::test_tool_reload_when_macro_is_altered
```

is the failing test case, unrelated by my reading of it.

**Pull request merged**